### PR TITLE
Add `TpmtSignature` and `TpmtSigScheme`

### PR DESCRIPTION
**Pull request by dzil123**

### CODE REVIEW

**Comment by dzil123** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by trent-reed** [comment]
Sorry for the delay here, been a mixture of vacation, power outages, sick, summits, and holidays all in one this past month. :) I will be reviewing this week.

File: base/src/lib.rs [hunk 1]
```diff
@@ -242,6 +242,23 @@ pub enum TpmiAlgRsaScheme{
...
 
+/// TpmiAlgSigScheme represents values that may appear in the scheme parameter of a TpmtSigScheme (TPMI_ALG_SIG_SCHEME).
+/// See definition in Part 2: Structures, section 9.37.
+#[open_enum]
+#[repr(u16)]
+#[rustfmt::skip] #[derive(Debug)] // Keep debug derivation separate for open_enum override.
+#[derive(Copy, Clone, PartialEq, Default, Marshalable)]
+#[allow(clippy::upper_case_acronyms)]
+pub enum TpmiAlgSigScheme {
+    RSAPSS = TpmAlgId::RSAPSS.0,
```
Comment (trent-reed) [inline base/src/lib.rs:253]: Our formatting of these things is inconsistent... Not your problem, just apologizing for the mess. Thanks for the PR!

**Comment by trent-reed** [comment]
Pretty straight-forward; thanks!

**Review by trent-reed — APPROVED** [review]

**Comment by oddcoder** [comment]
@trent-reed FYI, I think only us can merge so feel free to do so.

**Comment by trent-reed** [comment]
> @trent-reed FYI, I think only us can merge so feel free to do so.

Ah, I forgot - merging now, thanks!

**Comment by trent-reed** [comment]
The failed check is old code (near `derive_tpm_marshal`), unrelated to this change, I'll send another PR to fix.

**Pull request merged**